### PR TITLE
filter link checker output and retry for transient errors

### LINKED ISSUE #233: Please have `npm run test` print which test(s) failed
I sent a PR and the build failed and the log said:

```
Finished! 2147 links found. 1923 excluded. 1 broken.
```

BUT WHICH ONE?!?!?!

### PR DESCRIPTION
**Pull request by karthik2804**
closes #233 

Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: .build/check-broken-links.sh [hunk 1]
```diff
@@ -39,4 +42,25 @@ blc --recursive http://127.0.0.1:3000
...
+    | grep "├─BROKEN─" > broken_links || true
+
+if [ -s broken_links ]; then
+  echo "Some links are broken, retrying to check for transient errors"
+  while read -r line; do
+    url="$(echo $line | awk '{print $2}')"
+    if curl --retry 5 --retry-all-errors --retry-delay 1 --output /dev/null --silent --head --fail "$url"; then
+      echo "$url is not broken"
+    else
+      echo "$line" > final_broken
```
Comment (rajatjindal) [inline .build/check-broken-links.sh:55]: should be `>>`

Comment (karthik2804) [reply .build/check-broken-links.sh:55]: Good catch!

**Review by itowlson — APPROVED** [review]
@karthik2804 I'm not really qualified to review the script, but the functionality looks great - thanks!

**Review by rajatjindal — APPROVED** [review]
I wish we can do the retry with `blc` directly, but this is better then transient failures

**Pull request merged**